### PR TITLE
Untag dispatch find_first_of

### PR DESCRIPTION
**Pull request by AlexGuteniev**
Towards #189

### CODE REVIEW

File: stl/inc/xstring [hunk 1]
```diff
@@ -741,41 +741,36 @@ private:
...
-    if (_Needle_size != 0 && _Start_at < _Hay_size) { // room for match, look for it
-        const auto _End = _Haystack + _Hay_size;
-        for (auto _Match_try = _Haystack + _Start_at; _Match_try < _End; ++_Match_try) {
-            if (_Traits::find(_Needle, _Needle_size, *_Match_try)) {
-                return static_cast<size_t>(_Match_try - _Haystack); // found a match
+    const size_t _Needle_size) noexcept {
+    if constexpr (_Special) {
+        // in [_Haystack, _Haystack + _Hay_size), look for one of [_Needle, _Needle + _Needle_size), at/after _Start_at
+        // special case for std::char_traits
+        if (_Needle_size != 0 && _Start_at < _Hay_size) { // room for match, look for it
```
Comment (CaseyCarter) [inline stl/inc/xstring]: This is duplicated in both branches - we should push it out if the `if constexpr`.

**Review by CaseyCarter — APPROVED** [review]
Fix pushed.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for this series of great PRs! :cat: :cat2: :black_cat:

**Pull request merged**